documents/trail/ecs-mini-program.yml

```diff
@@ -0,0 +1,353 @@
+            Effect: Allow
+            Principal:
+              Service:
+                - ecs.aliyuncs.com
+      Policies:
+        - PolicyName:
+            'Fn::Sub':
+              - '${RamRoleName}Policy'
+              - RamRoleName:
+                  Ref: EcsRamRoleName
+          Description: 搭建支付宝小程序ECS实例角色权限策略
+          PolicyDocument:
+            Version: '1'
+            Statement:
+              - Action:
+                  - 'ecs:DescribeInstances'
+                  - 'ecs:DescribeInstanceStatus'
+                Effect: Allow
+                Resource:
+                  - '*'
+  EcsRamRoleAttachment:
+    Type: 'ALIYUN::ECS::RamRoleAttachment'
+    Properties:
+      InstanceIds:
+        - Ref: InstanceId
+      RamRoleName:
+        Ref: EcsRamRoleName
+    DependsOn: EcsRamRole
+  SecurityGroupIngress_80:
+    Type: 'ALIYUN::ECS::SecurityGroupIngress'
+    Properties:
+      SecurityGroupId:
+        'Fn::Jq':
+          - First
+          - '.[0].SecurityGroupIds[0]'
+          - 'Fn::GetAtt':
+              - DS_Instances
+              - Instances
+      SourceCidrIp: 0.0.0.0/0
+      IpProtocol: tcp
+      NicType: intranet
+      PortRange: 80/80
+  InstallWebServer:
+    Type: 'ALIYUN::ECS::RunCommand'
+    Properties:
+      InstanceIds:
+        - Ref: InstanceId
+      Type: RunShellScript
+      Sync: true
+      Timeout: 3600
+      CommandContent:
+        'Fn::Sub': >-
+          #!/bin/bash
+
+
+          if [ ! -f .ros.provision ]; then
+            echo "Name: 搭建支付宝小程序" > .ros.provision
+          fi
+
+
+
+          name=$(grep "^Name:" .ros.provision | awk -F':' '{print $2}' | tr -d
+          '[:space:]')
+
+
+          if [[ "$name" != "搭建支付宝小程序" ]]; then
+            echo "当前实例已使用过\"$name\"教程的一键配置，不能再使用本教程的一键配置"
+            ${WaitConditionHandle.CurlCli} --data-binary "{\"status\": \"FAILURE\", \"reason\": \"The current instance has already applied the configuration of the \\\"$name\\\" tutorial, and the configuration of this tutorial can no longer be applied.\"}"
+            exit 0
+          fi
+
+
+          if ! grep -q "^Step1: Install Python$" .ros.provision; then
+            echo "#########################"
+            echo "# Install Python"
+            echo "#########################"
+            mkdir /data && touch /data/GetServerInfo.py
+            cat > /data/GetServerInfo.py << EOF
+          # -*- coding: utf-8 -*-
+
+          from flask import Flask, jsonify, request
+
+          from aliyunsdkcore.client import AcsClient
+
+          from aliyunsdkcore.auth import credentials
+
+          import requests
+
+          import json
+
+          from aliyunsdkecs.request.v20140526 import DescribeInstancesRequest,
+          DescribeInstanceStatusRequest
+
+
+          app = Flask(__name__)
+
+
+          metaUrl =
+          'http://100.100.100.200/latest/meta-data/ram/security-credentials/${EcsRamRoleName}'
+
+          region = 'cn-beijing'
```

Comment:
If it is an ECS in another region, the returned information will be empty, and use the pseudo parameter  `ALIYUN::Region` instead